### DISCUSSION TITLE
platform/actors/development/builds-and-runs/state-persistence

### DISCUSSION THREAD

**Discussion by giscus** [apify/apify-docs#1147] [Comments]
# platform/actors/development/builds-and-runs/state-persistence

Maintain a long-running Actor's state to prevent unexpected restarts. See a code example on how to prevent a run in the case of a server shutdown.

https://docs.apify.com/platform/actors/development/builds-and-runs/state-persistence

**Comment by max4ever** [comment]
In order to be able to test/debug this event, could you show an example of how to emit this event? 
Thank you

**Reply by 0xsasky** [reply]
[Apify System Events (Programming Interface)](https://docs.apify.com/platform/actors/development/programming-interface/system-events)
